tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -120,6 +120,16 @@ internal void AddWafSecurityEvents(IReadOnlyCollection<object> events)
             _appSecRequestContext!.AddWafSecurityEvents(events);
         }
 
+        internal void AddStackTraceElement(Dictionary<string, object> stack, int maxStackTraces)
+        {
```

Comment:
maybe for another PR we do this for every method related to appsec in here, I wonder, if we shouldn't just instantiate an appsecrequest context at request's start to avoid volatile and interlock each time